core/jvm/src/test/scala/scalaz/zio/stream/StreamSpec.scala

```diff
@@ -374,4 +378,28 @@ class StreamSpec(implicit ee: org.specs2.concurrent.ExecutionEnv)
         l   <- ref.get
       } yield l.reverse must_=== (0 to 10).toList
     )
+
+  private def unTake =
+    unsafeRun(
+      Stream
+        .range(0, 10)
+        .toQueue[Nothing, Int](1)
+        .use { q =>
+          Stream.fromQueue(q).unTake.run(Sink.collect[Int])
+        }
+        .map(_ must_=== (0 to 10).toList)
+    )
+
+  private def unTakeError = {
+    val e = new RuntimeException("boom")
+    unsafeRunSync(
+      Stream
+        .range(0, 10)
+        .++(Stream.fail(e))
```

Comment:
Curious, why not write this as `(Stream.range(0, 10) ++ Stream.fail(e))`?

Reply:
It probably made sense before some sort of refactoring. ;-)